libs/nio/src/main/java/org/elasticsearch/nio/SocketChannelContext.java

```diff
@@ -127,7 +127,7 @@ public boolean connect() throws IOException {
     }
 
     public void sendMessage(Object message, BiConsumer<Void, Exception> listener) {
-        if (isClosing.get()) {
+        if (selectorShouldClose()) {
```

Comment:
This change should be reverted. `selectorShouldClose`  should not be accessed from outside the event loop thread. `sendMessage` can be called from any thread.

Reply:
reverted